metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/http/AbstractHttpClient.java

```diff
@@ -106,71 +79,139 @@ private void cleanConnections(PoolingHttpClientConnectionManager connectionManag
+    CancelableBodyWrapper<InputStream> bodyWrapper = new CancelableBodyWrapper<>(handler);
+
+     httpResponse = makeHttpRequest(resourceUlr, bodyWrapper);
 
     // Set up the abort trigger
     final TimerTask abortTask = new TimerTask() {
       @Override
       public void run() {
-        LOGGER.info("Aborting request due to time limit: {}.", resourceUlr);
-        httpGet.abort();
+        LOGGER.info("Aborting request due to time limit: {}.", resourceUlr.getPath());
+        bodyWrapper.cancel();
+        if (httpResponse.body() != null) {
+          try {
+            httpResponse.body().close();
+          } catch (IOException e) {
+            LOGGER.warn(
+                    "Something went wrong while trying to close the input stream after cancelling the http request.",
+                    e);
+          }
+        }
       }
     };
     final Timer timer = new Timer(true);
     timer.schedule(abortTask, requestTimeout);
 
-    // Execute the request.
-    try (final CloseableHttpResponse responseObject = client.execute(httpGet, context)) {
-
-      // Do first analysis
-      final HttpEntity responseEntity = performThrowingFunction(responseObject, response -> {
-        final int status = response.getCode();
-        if (!httpCallIsSuccessful(status)) {
-          throw new IOException("Download failed of resource " + resourceUlr + ". Status code " +
-                  status + " (message: " + response.getReasonPhrase() + ").");
-        }
-        return response.getEntity();
-      });
+    final int statusCode;
+
+    if (httpResponse != null) {
+      statusCode = httpResponse.statusCode();
+    } else {
+      throw new IOException("A problem occurred when sending the request");
+    }
+
+    final Optional<String> redirectUris = httpResponse.headers().firstValue("Location");
+    final URI actualUri;
+
+    // Do first check redirection and analysis
+    if(Family.familyOf(statusCode) == Family.REDIRECTION){
+      httpResponse = performRedirect(statusCode, resourceUlr.resolve(redirectUris.get()),
+          maxNumberOfRedirects, bodyWrapper);
+      actualUri = httpResponse.uri();
+      if(actualUri == null){
+        throw new IOException("There was some trouble retrieving the uri");
+      }
+    } else if (Status.fromStatusCode(statusCode) != Status.OK) {
+      throw new IOException(
+          "Download failed of resource " + resourceUlr + ". Status code " + statusCode);
+    } else {
+      actualUri = httpResponse.uri();
+    }
+    // Obtain header information.
+    final Optional<String> mimeType = httpResponse.headers().firstValue("Content-Type");
+    final long fileSize = httpResponse.headers().firstValueAsLong("Content-Length").orElse(0);
+
+    // Process the result.
+    final ContentRetriever content = httpResponse.body() == null ?
+        ContentRetriever.forEmptyContent() : httpResponse::body;
+    final R result;
+    try {
+      result = createResult(resourceEntry, actualUri, mimeType.orElse(null),
```

Comment:
Same here, what if mimeType is blank, are we ok with that?

Reply:
Not quite sure if `mimeType` being blank scenario is handled in `createResult` method. I did check with Jochen if it was ok to pass a `null` `mimeType`, for which the method does handle if the value is `null`. 
I'll add a `isBlank` check